migration.md

```diff
@@ -0,0 +1,301 @@
+# Migration Guide From v4 to v5
+
+## Summary
+
+After upgrading from v4 to v5, there are **TWO** major issues that we need to resolve.
+
+1. **Compiler complain that `Defaults.Key<Value>` is not conform to `Defaults.Serializable`.**  
+
+	Please follow the steps below:
+	- [From `Codable` struct in Defaults v4 to `Codable` struct in Defaults v5](#from-codable-struct-in-defaults-v4-to-codable-struct-in-defaults-v5)
+	- [From `Codable` enum in Defaults v4 to `Codable` enum in Defaults v5](#from-codable-enum-in-defaults-v4-to-codable-enum-in-defaults-v5)
+
+
+2. **Previous value in UserDefaults is not readable. (ex. `Defaults[.array]` return `null`).**
+
+	Please follow the steps below:
+	- [From `Codable Array/Dictionary/Set` in Defaults v4 to `Native Array/Dictionary/Set`(With Native Supported Elements) in Defaults v5](#from-codable-arraydictionaryset-in-defaults-v4-to-native-arraydictionarysetwith-native-supported-elements-in-defaults-v5)
+	- [From `Codable Array/Dictionary/Set` in Defaults v4 to `Native Array/Dictionary/Set`(With Codable Elements) in Defaults v5](#from-codable-arraydictionaryset-in-defaults-v4-to-native-arraydictionarysetwith-codable-elements-in-defaults-v5)  
+
+	**Caution:**
+	- This is a breaking change, there is no way to convert it back to `Codable Array/Dictionary/Set` so far.
+	- If the migration is not successful or incomplete. Edit `Defaults.Key` might cause data loss.
+
+
+- **Optional migration**
+
+	Defaults also provide a migration guide to let users convert them `Codable` things into the UserDefaults NativeSupported type, but it's optional.
+
+	- [From `Codable` enum in Defaults v4 to `RawRepresentable` in Defaults v5](#from-codable-enum-in-defaults-v4-to-rawrepresentable-in-defaults-v5-optional)
+	- [From `Codable` struct in Defaults v4 to `Dictionary` in Defaults v5](#from-codable-struct-in-defaults-v4-to-dictionary-in-defaults-v5-optional)
+
+---
+
+### From `Codable` struct in Defaults v4 to `Codable` struct in Defaults v5
+
+Before v4, `struct` have to conform to protocol `Codable` to store it as a JSON string.  
+
+After v5, `struct` have to conform to protocol `Defaults.Serializable & Codable` to store it as a JSON string.  
+
+#### Before migration, your code should be like this
+```swift
+private struct TimeZone: Codable {
+	var id: String
+	var name: String
+}
+
+extension Defaults.Keys {
+	static let timezone = Defaults.Key<TimeZone?>("TimeZone")
+}
+```
+
+#### Migration steps
+1. Let `TimeZone` protocol conform to `Defaults.Serializable`.
+```swift
+private struct TimeZone: Defaults.Serializable & Codable {
```

Comment:
```suggestion
private struct TimeZone: Defaults.Serializable, Codable {
```

Same applies to all the others. They are equivalent, but I prefer the comma version.